src/Microsoft.DotNet.TestFramework/Microsoft.DotNet.TestFramework.TestScenario.cs

```
@@ -0,0 +1,115 @@
+            Projects = GetAllProjects(SourceRoot);
+
+            if (!skipRestore)
+            {
+                Restore();
+            }
+
+            if (!skipBuild)
+            {
+                Build();
+            }
+        }
+
+        private IEnumerable<string> Projects
+        {
+            get; set;
+        }
+
+        public string SourceRoot
+        {
+            get;
+            private set;
+        }
+
+        public static TestScenario Create(string testSourceRoot, bool skipRestore = false, bool skipBuild = false)
+        {
+            TestScenario testScenario;
+            lock (_scenarioCache)
+            {
+                if (!_scenarioCache.TryGetValue(testSourceRoot, out testScenario))
+                {
+                    testScenario = new TestScenario(testSourceRoot, skipRestore, skipBuild);
+                    _scenarioCache.Add(testSourceRoot, testScenario);
+                }
+            }
+
+            return testScenario;
+        }
+
+        public TestInstance CreateTestInstance([CallerMemberName] string callingMethod = "", string identifier = "")
+        {
+            string projectName = new DirectoryInfo(SourceRoot).Name;
+            string testDestination = Path.Combine(AppContext.BaseDirectory, callingMethod + identifier, projectName);
+            var testInstance = new TestInstance(this, testDestination);
+            return testInstance;
+        }
+
+        internal void Build()
+        {
+            foreach (var project in Projects)
+            {
+                string[] buildArgs = new string[] { "build", project };
+                var commandResult = Command.Create("dotnet", buildArgs)
+                                        .CaptureStdOut()
+                                        .CaptureStdErr()
+                                        .Execute();
+
+                Console.WriteLine(commandResult.StdOut);
+                Console.WriteLine(commandResult.StdErr);
```

This will be noisy. Can we swallow unless error?